tests/src/core/testqgsexpression.cpp

```diff
@@ -607,6 +607,9 @@ class TestQgsExpression: public QObject
       QTest::newRow( "set color part yellow" ) << "to_int(color_part(set_color_part(color_cmyk(21,0,92,70),'yellow',96),'yellow'))" << false << QVariant( 96 );
       QTest::newRow( "set color part black" ) << "to_int(color_part(set_color_part(color_cmyk(21,0,92,70),'black',100),'black'))" << false << QVariant( 100 );
 
+      QTest::newRow( "color darker" ) << "darker('200,100,30',150)" << false << QVariant( "133,66,20,255" );
+      QTest::newRow( "color lighter" ) << "lighter('200,100,30',150)" << false << QVariant( "255,154,83,255" );
```

Comment:
@nyalldawson I thought since color_part already has a test covering non-color string, and that the lighter / darker functions create a QColor object in the exact same way, it'd be a duplication. Should I still go ahead and add such a test?